src/audio/mixin_mixout/mixin_mixout_hifi3.c

```diff
@@ -593,4 +596,171 @@ const struct mix_func_map mix_func_map[] = {
+		      const struct audio_stream __sparse_cache **sources, uint32_t num_sources,
+		      uint32_t frames)
+{
+	ae_int32x2 *in[PLATFORM_MAX_CHANNELS];
+	ae_int32x2 *out = audio_stream_get_wptr(sink);
+	ae_int32x2 val;
+	ae_int32x2 sample = AE_ZERO32();
+	unsigned int n, m, nmax, i, j, left_samples;
+	unsigned int samples = frames * audio_stream_get_channels(sink);
+
+	for (j = 0; j < num_sources; j++)
+		in[j] = audio_stream_get_rptr(sources[j]);
+
+	for (left_samples = samples; left_samples; left_samples -= n) {
+		out = audio_stream_wrap(sink,  out);
+		nmax = audio_stream_samples_without_wrap_s32(sink, out);
+		n = MIN(left_samples, nmax);
+		for (j = 0; j < num_sources; j++) {
+			in[j] = audio_stream_wrap(sources[j], in[j]);
+			nmax = audio_stream_samples_without_wrap_s32(sources[j], in[j]);
+			n = MIN(n, nmax);
+		}
+		m = n >> 1;
+		for (i = 0; i < m; i++) {
+			val = AE_ZERO32();
+			for (j = 0; j < num_sources; j++) {
+				/* load two 32 bit samples, 8 is sizeof(ae_int32x2) */
+				AE_L32X2_IP(sample, in[j], 8);
+				/* Sign extend */
+				sample = AE_SRAA32RS(AE_SLAI32(sample, 8), 8);
+				val = AE_ADD32S(val, sample);
+			}
+			/*Saturate to 24 bits */
+			val = AE_SRAA32S(AE_SLAA32S(val, 8), 8);
+
+			/* store two 32 bit samples, 8 is sizeof(ae_int32x2) */
+			AE_S32X2_IP(val, out, 8);
+		}
+	}
+}
+#endif /* CONFIG_FORMAT_S24LE */
+
+#if CONFIG_FORMAT_S32LE
+/* Mix n 32 bit PCM source streams to one sink stream */
+static void mix_n_s32(struct comp_dev *dev, struct audio_stream __sparse_cache *sink,
+		      const struct audio_stream __sparse_cache **sources, uint32_t num_sources,
+		      uint32_t frames)
+{
+	ae_q32s * in[PLATFORM_MAX_CHANNELS];
+	ae_int32 *out = audio_stream_get_wptr(sink);
+	ae_int64 sample;
+	ae_int64 val;
+	ae_int32x2 res;
+	unsigned int n, nmax, i, j, left_samples;
+	unsigned int m = 0;
+	unsigned int samples = frames * audio_stream_get_channels(sink);
+
+	for (j = 0; j < num_sources; j++)
+		in[j] = audio_stream_get_rptr(sources[j]);
+
+	for (left_samples = samples; left_samples; left_samples -= n) {
+		out = audio_stream_wrap(sink,  out);
+		nmax = audio_stream_samples_without_wrap_s32(sink, out);
+		n = MIN(left_samples, nmax);
+		for (j = 0; j < num_sources; j++) {
+			in[j] = audio_stream_wrap(sources[j], in[j] + m);
+			nmax = audio_stream_samples_without_wrap_s32(sources[j], in[j]);
+			n = MIN(n, nmax);
+		}
+		/*record the processed samples for next address iteration */
+		m = n;
+		for (i = 0; i < m; i++) {
+			val =  AE_ZERO64();
+			for (j = 0; j < num_sources; j++) {
+				/* load one 32 bit sample */
+				sample = AE_L32M_X(in[j],  i * sizeof(ae_q32s));
+				val = AE_ADD64S(val, sample);
+			}
+			/*Saturate to 32 bits */
+			res =  AE_ROUND32X2F48SSYM(val, val);
+
+			/* store one 32 bit samples */
+			AE_S32_L_IP(res, out, sizeof(ae_int32));
+		}
+	}
+}
+#endif /* CONFIG_FORMAT_S32LE */
+
+const struct mixer_func_map mixer_func_map[] = {
```

Comment:
this isn't used anywhere yet, right? So this PR is so far incomplete?

Reply:
line 490

Reply:
> 

hmm on the other hand the mix_func_map is what is unused with my PR and this mixer_func_map is used. So the PR is incomplete only in terms of getting all tests to pass and removing unused code